character_map.h

```diff
@@ -0,0 +1,91 @@
+/*
+* Copyright:  Copyright Dimakopoulos Theodoros 2019
+* License:    $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+* Authors:    Dimakopoulos Theodoros
+*/
+#pragma once

```

Comment:
"#pragma once" isn't the best practice. The ifndef method is more preferred. Like:
```C
#ifndef CHARACTER_MAP_H
#define CHARACTER_MAP_H

// Code...

#endif //CHARACTER_MAP_H```

Reply:
It's fine. First of all, please when we state that X is not good and Y is better, we should state both why X is not and good _and_ why Y is better. :)
So, basically `#pragma once` is not "standard C++" (i.e. a thing that all C++ compilers have to do the exact same thing) and as all `pragma`s, it depends on how the compiler interprets it. That said, this project does not a hard constraint to be standard C++ when it doesn't have to. In that case, it doesn't have, as any modern compiler treats this as a normal guard.
And we avoid introducing a useless `#define` like `CHARACTER_MAP_H`.

All in all, I don't think there's a right or wrong way in this case, that's why I haven't commented on this in any other PR.

_Note_: I'm glad when other contributors of the project review, but please always have in mind that it's pretty important to explain in depth all the decisions made (or proposed).